### ISSUE TITLE
Is there an option to configure custom endpoint to TF acceptance tests?

### ISSUE THREAD

**Issue by romfreiman** [hashicorp/terraform-provider-aws#7739]
I want to be albe to run TF provider tests not using aws but local custom endpoints. Is there such option?

AWS_ACCESS_KEY_ID=*** AWS_SECRET_ACCESS_KEY=***  TF_ACC=1 go test ./aws -v -timeout 120m

I would like to configure ec2 / elb/ cloudwatch customer endpoints instead of accessing aws.

**Comment by rom-stratoscale** [comment]
?

**Comment by bflad** [comment]
Hi @rom-stratoscale 👋 We have not optimized the testing or provider configuration beyond just working with public AWS partitions such as Standard, GovCloud (US), and China, but we may do so in the future to more easily support Localstack and other non-public AWS partitions.

You may be able to workaround this by manually inserting a provider configuration within each test configuration as necessary though, e.g.

```hcl
provider "aws" {
  endpoints {
    apigateway     = "http://localhost:4567"
    cloudformation = "http://localhost:4581"
    cloudwatch     = "http://localhost:4582"
    dynamodb       = "http://localhost:4569"
    es             = "http://localhost:4578"
    iam            = "http://localhost:4593"
    kinesis        = "http://localhost:4568"
    lambda         = "http://localhost:4574"
    r53            = "http://localhost:4580"
    s3             = "http://localhost:4572"
    sns            = "http://localhost:4575"
    sqs            = "http://localhost:4576"
    ssm            = "http://localhost:4583"
    sts            = "http://localhost:4592"
  }
}

# ... other existing test configuration ...
```

**Comment by rom-stratoscale** [comment]
Thanks @bflad 
Just to be clear, I should adjust the configuration in similar place to this example, correct?
https://github.com/terraform-providers/terraform-provider-aws/blob/5a062fcdc5b24d82017aa15bc561fac1f152a2c6/aws/resource_aws_ami_from_instance_test.go#L145

Any idea how should I actually debug where the requests goes? Tried tcpdump / netstat - but could not find any connection to my endpoint.

Thanks

**Comment by rom-stratoscale** [comment]
Also, from what I understand, the precheck (testAccPreCheck) does not load the test config but uses the defaults (not sure where defined) - does it mean that it will go no matter what to AWS?

**Comment by rom-stratoscale** [comment]
Update: So, after some digging into the code, apparently there are 2 flows - one TF and one aws sdk.
I managed to make TF work with custom endpoints by sed-ing the custom enpoints to relevant tests.
But.. not sure how to update the Config so aws sdk will read it and work with the same custom enpoints.
I tried to update ~/.aws/config (with enpoint plugin) but seems that this file is not used by the sdk. 
@bflad any other suggestion?
